app/models/developers/external_profiles/linkedin_profile_fetcher.rb

```diff
@@ -1,44 +1,46 @@
 module Developers::ExternalProfiles
   class LinkedinProfileFetcher
     def developer_profiles
-      profiles = []
       developers = developers_with_linkedin_profiles
       Rails.logger.info "Fetching #{developers.count} #{"LinkedIn profile".pluralize(developers.count)}..."
 
       developers.find_each.with_index do |developer, index|
         response = fetch_linkedin_profile(developer.linkedin)
-        record = external_profile(developer, response)
-        profiles << record if record.present?
+        external_profile = Developers::ExternalProfile.find_or_initialize_by(developer_id: developer.id, site: "linkedin")
+        record = external_profile(developer, response, external_profile)
+        if record.present?
+          external_profile.data = record[:data]
+          external_profile.error = record[:error]
+          external_profile.fetched_at = Time.now
+          external_profile.save!
+        else
+          external_profile.fetched_at = Time.now
+          external_profile.save(touch: false)
```

Comment:
If data is fetched for developer after days from its last fetched_date but if there is no change in its API data in that case we are just updated fetched_at attribute to current date but not updating updated_at as we don't need this developer data in email sent for profiles updated in last 30 days.